Sources/Crust/Shine.swift

```diff
@@ -18,38 +18,99 @@ private enum LayoutBlock {
   }
 }
 
-fileprivate extension TokenSyntax {
-  func hasEquivalentLeadingWhitespace(to other: TokenSyntax) -> Bool {
-    let selftrivia = self.leadingTrivia.filter({ if case .comment(_) = $0 { return false }; return true })
-    let othertrivia = other.leadingTrivia.filter({ if case .comment(_) = $0 { return false }; return true })
-    guard selftrivia.count == othertrivia.count else {
-      return false
+struct WhitespaceSummary {
+  enum Spacer: Equatable {
+    case spaces(Int)
+    case tabs(Int)
+
+    static func == (lhs: Spacer, rhs: Spacer) -> Bool {
+      switch (lhs, rhs) {
+      case let (.spaces(l), .spaces(r)): return l == r
+      case let (.tabs(l), .tabs(r)): return l == r
+      default: return false
+      }
     }
-    
-    for (l, r) in zip(selftrivia.reversed(), othertrivia.reversed()) {
-      switch (l, r) {
-      // Search ends at the nearest newline.
-      case (.newlines(_), .newlines(_)):
-        return true
-      // Obvious mismatches between spaces and tabs
-      case (.spaces(_), .tabs(_)):
-        return false
-      case (.tabs(_), .spaces(_)):
-        return false
-      case (.spaces(let ln), .spaces(let rn)) where ln != rn:
-        return false
-      case (.tabs(let ln), .tabs(let rn)) where ln != rn:
-        return false
-      // Otherwise we have equal numbers of spaces or tabs
-      case (.spaces(_), .spaces(_)):
+  }
+  let sequence: [Spacer]
+  let totals: (Int, Int)
+  let hasNewline: Bool
+
+  init(_ t : Trivia) {
+    var seq = [Spacer]()
+    var spaces = 0
+    var tabs = 0
+    var newl = false
+    for i in (0..<t.count).reversed() {
+      switch t[i] {
+      case .spaces(let ss):
+        spaces += ss
+        seq.append(.spaces(ss))
         continue
-      case (.tabs(_), .tabs(_)):
+      case .tabs(let ts):
+        tabs += ts
+        seq.append(.tabs(ts))
         continue
-      // Keep searching
-      default:
+      case .comment(_):
         continue
+      default:
+        newl = true
+        break
+      }
+      break
```

Comment:
Is this correct?

Reply:
It iterates in reverse until the nearest non-tab or non-space and records that there was a newline. Odd, but correct.

Reply:
Ah I wasn't considering the `continue`s. This is fine.